### PR TITLE
UIMollierObjects form updated and refactoring

### PR DESCRIPTION
**Pull request by MaciejDengusiak**
### Issues addressed by this PR

Fixes #

### CODE REVIEW

File: Grasshopper/SAM.Core.Mollier.UI.Grasshopper/Component/SAMMollierCreateUndefinedProcess.cs [hunk 1]
```diff
@@ -57,7 +57,7 @@ protected override GH_SAMParam[] Outputs
             get
             {
                 List<GH_SAMParam> result = new List<GH_SAMParam>();
-                result.Add(new GH_SAMParam(new GooMollierProcessParam() { Name = "undefinedProcess", NickName = "undefinedProcess", Description = "Undefined Process", Access = GH_ParamAccess.item }, ParamVisibility.Binding));
+                result.Add(new GH_SAMParam(new GooMollierProcessParam() { Name = "RoomProcess", NickName = "RoomProcess", Description = "Undefined Process", Access = GH_ParamAccess.item }, ParamVisibility.Binding));
```
Comment (ZiolkowskiJakub) [inline Grasshopper/SAM.Core.Mollier.UI.Grasshopper/Component/SAMMollierCreateUndefinedProcess.cs:60]: RoomProcess -> roomProcess

File: Grasshopper/SAM.Core.Mollier.UI.Grasshopper/Component/SAMMollierCreateUndefinedProcess.cs [hunk 2]
```diff
@@ -133,13 +133,13 @@ protected override void SolveInstance(IGH_DataAccess dataAccess)
                 dataAccess.GetData(index, ref endLabel);
             }
 
-            UndefinedProcess undefinedProcess = Core.Mollier.Create.UndefinedProcess(start, end);
+            RoomProcess RoomProcess = Core.Mollier.Create.RoomProcess(start, end);
```
Comment (ZiolkowskiJakub) [inline Grasshopper/SAM.Core.Mollier.UI.Grasshopper/Component/SAMMollierCreateUndefinedProcess.cs:136]: RoomProcess -> roomProcess

File: Grasshopper/SAM.Core.Mollier.UI.Grasshopper/Component/SAMMollierCreateUndefinedProcess.cs [hunk 3]
```diff
@@ -133,13 +133,13 @@ protected override void SolveInstance(IGH_DataAccess dataAccess)
...
 
 
-            index = Params.IndexOfOutputParam("undefinedProcess");
+            index = Params.IndexOfOutputParam("RoomProcess");
```
Comment (ZiolkowskiJakub) [inline Grasshopper/SAM.Core.Mollier.UI.Grasshopper/Component/SAMMollierCreateUndefinedProcess.cs:139]: RoomProcess -> roomProcess

File: Grasshopper/SAM.Core.Mollier.UI.Grasshopper/Component/SAMMollierCreateUndefinedProcessByEpsilon.cs [hunk 4]
```diff
@@ -71,7 +71,7 @@ protected override GH_SAMParam[] Outputs
             get
             {
                 List<GH_SAMParam> result = new List<GH_SAMParam>();
-                result.Add(new GH_SAMParam(new GooMollierProcessParam() { Name = "undefinedProcess", NickName = "undefinedProcess", Description = "Undefined Process", Access = GH_ParamAccess.item }, ParamVisibility.Binding));
+                result.Add(new GH_SAMParam(new GooMollierProcessParam() { Name = "RoomProcess", NickName = "RoomProcess", Description = "Undefined Process", Access = GH_ParamAccess.item }, ParamVisibility.Binding));
```
Comment (ZiolkowskiJakub) [inline Grasshopper/SAM.Core.Mollier.UI.Grasshopper/Component/SAMMollierCreateUndefinedProcessByEpsilon.cs:74]: RoomProcess -> roomProcess

File: Grasshopper/SAM.Core.Mollier.UI.Grasshopper/Component/SAMMollierCreateUndefinedProcessByEpsilon.cs [hunk 5]
```diff
@@ -163,13 +163,13 @@ protected override void SolveInstance(IGH_DataAccess dataAccess)
                 dataAccess.GetData(index, ref endLabel);
             }
 
-            UndefinedProcess undefinedProcess = Mollier.Create.UndefinedProcess_ByEpsilonAndHumidityRatioDifference(mollierPoint, epsilon, (humidityRatio / 1000) - mollierPoint.HumidityRatio, start);
+            RoomProcess RoomProcess = Mollier.Create.RoomProcess_ByEpsilonAndHumidityRatioDifference(mollierPoint, epsilon, (humidityRatio / 1000) - mollierPoint.HumidityRatio, start);
 
 
-            index = Params.IndexOfOutputParam("undefinedProcess");
+            index = Params.IndexOfOutputParam("RoomProcess");
```
Comment (ZiolkowskiJakub) [inline Grasshopper/SAM.Core.Mollier.UI.Grasshopper/Component/SAMMollierCreateUndefinedProcessByEpsilon.cs:169]: RoomProcess -> roomProcess

File: Mollier/SAM.Core.Mollier.UI/Classes/DisplayAnalyticalObject.cs [hunk 6]
```diff
@@ -8,21 +8,24 @@ namespace SAM.Core.Mollier.UI
     public class DisplayUIMollierObject
     {
         public IUIMollierObject UIMollierObject { get; }
+        public string GroupName { get; }
         public int Index { get; }
         private double airFlow;
         private Units.UnitType airFlowUnit;
 
-        public DisplayUIMollierObject(UIMollierPoint uIMollierPoint)
+        public DisplayUIMollierObject(UIMollierPoint uIMollierPoint, string groupName = "")
```
Comment (ZiolkowskiJakub) [inline Mollier/SAM.Core.Mollier.UI/Classes/DisplayAnalyticalObject.cs:16]: use string.Empty instead ""

File: Mollier/SAM.Core.Mollier.UI/Classes/DisplayAnalyticalObject.cs [hunk 7]
```diff
@@ -8,21 +8,24 @@ namespace SAM.Core.Mollier.UI
...
         {
             UIMollierObject = uIMollierPoint;
+            GroupName = groupName;
         }
 
-        public DisplayUIMollierObject(UIMollierProcess uIMollierProcess, int index, double airFlow, Units.UnitType airFlowUnit)
+        public DisplayUIMollierObject(UIMollierProcess uIMollierProcess, int index, double airFlow, Units.UnitType airFlowUnit, string groupName = "")
```
Comment (ZiolkowskiJakub) [inline Mollier/SAM.Core.Mollier.UI/Classes/DisplayAnalyticalObject.cs:22]: use string.Empty instead ""

File: Mollier/SAM.Core.Mollier.UI/Forms/MollierForm.cs [hunk 8]
```diff
@@ -1057,13 +1057,13 @@ private void AddProcess_ByEpsilonAndHumidityRatioDifference(MollierPointSelected
                 return;
             }
 
-            UndefinedProcess undefinedProcess = Mollier.Create.UndefinedProcess_ByEpsilonAndHumidityRatioDifference(mollierPoint, epsilon, (humidityRatio / 1000) - mollierPoint.HumidityRatio);
-            if (undefinedProcess == null)
+            RoomProcess RoomProcess = Mollier.Create.RoomProcess_ByEpsilonAndHumidityRatioDifference(mollierPoint, epsilon, (humidityRatio / 1000) - mollierPoint.HumidityRatio);
```
Comment (ZiolkowskiJakub) [inline Mollier/SAM.Core.Mollier.UI/Forms/MollierForm.cs:1060]: RoomProcess -> roomProcess

File: Mollier/SAM.Core.Mollier.UI/Query/GroupName.cs [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+using System.Collections.Generic;
+using System.Windows.Forms.DataVisualization.Charting;
+
+namespace SAM.Core.Mollier.UI
+{
+    public static partial class Query
+    {
+        public static string GroupName(IMollierObject mollierObject, IEnumerable<MollierGroup> mollierGroups)
+        {
+            string result = "";
```
Comment (ZiolkowskiJakub) [inline Mollier/SAM.Core.Mollier.UI/Query/GroupName.cs:11]: use string.Empty instead ""

**Review by ZiolkowskiJakub — CHANGES_REQUESTED** [review]

File: Grasshopper/SAM.Core.Mollier.UI.Grasshopper/Component/SAMMollierCreateUndefinedProcess.cs (hunk 1 above)
Comment (MaciejDengusiak) [reply Grasshopper/SAM.Core.Mollier.UI.Grasshopper/Component/SAMMollierCreateUndefinedProcess.cs:60]: Done

File: Grasshopper/SAM.Core.Mollier.UI.Grasshopper/Component/SAMMollierCreateUndefinedProcess.cs (hunk 3 above)
Comment (MaciejDengusiak) [reply Grasshopper/SAM.Core.Mollier.UI.Grasshopper/Component/SAMMollierCreateUndefinedProcess.cs:139]: Done

File: Mollier/SAM.Core.Mollier.UI/Classes/DisplayAnalyticalObject.cs (hunk 6 above)
Comment (MaciejDengusiak) [reply Mollier/SAM.Core.Mollier.UI/Classes/DisplayAnalyticalObject.cs:16]: Done

File: Grasshopper/SAM.Core.Mollier.UI.Grasshopper/Component/SAMMollierCreateUndefinedProcessByEpsilon.cs (hunk 5 above)
Comment (MaciejDengusiak) [reply Grasshopper/SAM.Core.Mollier.UI.Grasshopper/Component/SAMMollierCreateUndefinedProcessByEpsilon.cs:169]: Done

File: Mollier/SAM.Core.Mollier.UI/Forms/MollierForm.cs (hunk 8 above)
Comment (MaciejDengusiak) [reply Mollier/SAM.Core.Mollier.UI/Forms/MollierForm.cs:1060]: Done

File: Mollier/SAM.Core.Mollier.UI/Query/GroupName.cs (hunk 9 above)
Comment (MaciejDengusiak) [reply Mollier/SAM.Core.Mollier.UI/Query/GroupName.cs:11]: Done

**Review by ZiolkowskiJakub — APPROVED** [review]

**Pull request merged**